ge/org.osate.ge.errormodel/src/org/osate/ge/errormodel/ErrorPropagationHandler.java

```diff
@@ -0,0 +1,164 @@
+		segments[segmentIndex++] = ctx.getReferenceBuilder().getCanonicalReference(bo.getContainingClassifier())
+				.encode();
+
+		if (bo.getKind() == null) {
+			for (FeatureorPPReference t = bo.getFeatureorPPRef(); t != null; t = t.getNext()) {
+				final String name = t.getFeatureorPP() == null ? null : t.getFeatureorPP().getName();
+				segments[segmentIndex++] = Strings.isNullOrEmpty(name) ? "?" : name;
+			}
+		} else {
+			segments[segmentIndex++] = bo.getKind();
+		}
+
+		segments[segmentIndex++] = Boolean.toString(bo.isNot());
+		segments[segmentIndex++] = bo.getDirection() == null ? "<null>" : bo.getDirection().getLiteral();
+
+		// Create reference
+		return new CanonicalBusinessObjectReference(segments);
+	}
+
+	@Override
+	public RelativeBusinessObjectReference getRelativeReference(final ReferenceContext ctx) {
+		final ErrorPropagation bo = ctx.getBusinessObject(ErrorPropagation.class).get();
+		return ErrorModelReferenceUtil.getRelativeReferenceForPropagation(bo.isNot(),
+				bo.getDirection());
+	}
+
+	@Override
+	public boolean canDelete(final CanDeleteContext ctx) {
+		final ErrorPropagation bo = ctx.getBusinessObject(ErrorPropagation.class).get();
+
+		// Don't allow deleting if there exists an error flow that references the error propagation
+		final CombinedErrorModelSubclause combined = CombinedErrorModelSubclause.create(bo.getContainingClassifier());
+		return !combined.getFlows().anyMatch(f -> {
+			if (f instanceof ErrorSource) {
+				final ErrorSource src = (ErrorSource) f;
+				return src.getSourceModelElement() == bo;
+			} else if (f instanceof ErrorSink) {
+				final ErrorSink snk = (ErrorSink) f;
+				return snk.getIncoming() == bo;
+			} else if (f instanceof ErrorPath) {
+				final ErrorPath path = (ErrorPath) f;
+				return path.getIncoming() == bo || path.getOutgoing() == bo;
+			}
+			return false;
+		});
+	}
+
+	@Override
+	public Optional<GraphicalConfiguration> getGraphicalConfiguration(final GetGraphicalConfigurationContext ctx) {
+		// Determine style
+		final StyleBuilder sb = StyleBuilder.create(GraphicalExtensionUtil.isInherited(ctx.getBusinessObjectContext())
+				? GraphicalExtensionUtil.STYLE_INHERITED_ELEMENT
+				: Style.EMPTY).fontSize(8.0);
+
+		return Optional
+				.of(GraphicalConfigurationBuilder.create().graphic(graphic).style(sb.build()).decoration().build());
+	}
+
+	@Override
+	public String getName(final GetNameContext ctx) {
+		return ctx.getBusinessObject(ErrorPropagation.class)
+				.map(bo -> (bo.isNot() ? "not " : "") + (bo.getDirection() == null ? "" : bo.getDirection()))
+				.orElse("");
+	}
+
+	@Override
+	public String getNameForDiagram(final GetNameForDiagramContext ctx) {
+		return ctx.getBusinessObjectContext().getBusinessObject(ErrorPropagation.class)
+				.map(bo -> (bo.isNot() ? " not " : "") + (bo.getDirection() == null ? "" : (bo.getDirection() + " "))
+						+ "propagation " + getLabelForTypeSet(bo.getTypeSet()))
```

Comment:
I don't know if I like the label showing the type set.  It can get very long and doesn't seem like there is a way to hide the label.  In similar cases like with the Error Type Set element, the user can just click on the element and view the type set in the Properties View.  I think I like that approach where the user can click on the "out propagation" or "in propagation" label, then view the type set in the Properties View.  The Properties View already shows the type set for the propagations.

Just my opinion, I can be swayed on this stance, just something to think about.